primitives/src/predicate.rs

```diff
@@ -108,6 +130,26 @@ pub fn run(rule: &Rule, context: &Context) -> bool {
         RuleType::IsAbsent(ref claim) => not(exists(claim)).evaluate(context),
         RuleType::IsAnyOf(ref claims) => any(claims).evaluate(context),
         RuleType::IsNoneOf(ref claims) => not(any(claims)).evaluate(context),
+        RuleType::IsIdentity(ref id) => {
+            equals(id, &context.treasury.unwrap_or_default()).evaluate(context)
+        }
+    }
+}
+
+// MatchPredicate
+// ======================================================
+
+/// It matches `id` with treasury in the context.
+#[derive(Clone, Debug)]
+pub struct MatchPredicate<'a> {
```

Comment:
`Match` sounds a bit generic. It it makes a statement about the treasury identity, maybe replace `Match` with `Treasury` or `TreasuryIdentity`?